tests/Doctrine/Migrations/Tests/Tools/Console/Command/MigrateCommandTest.php

```diff
@@ -74,63 +75,121 @@ class MigrateCommandTest extends MigrationTestCase
 
     public function testTargetUnknownVersion() : void
     {
-        $result = new ExecutionResult(new Version('A'));
-        $this->storage->complete($result);
-
         $this->migrateCommandTester->execute(
-            ['version' => 'A'],
+            ['version' => 'B'],
             ['interactive' => false]
         );
 
-        self::assertStringContainsString('[ERROR] Unknown version: A', $this->migrateCommandTester->getDisplay(true));
+        self::assertStringContainsString('[ERROR] Unknown version: B', $this->migrateCommandTester->getDisplay(true));
         self::assertSame(1, $this->migrateCommandTester->getStatusCode());
     }
 
+    /**
+     * @return bool[][]
+     */
+    public function getMigrateWithMigrationsOrWithout() : array
+    {
+        return [
+            // migrations available, allow-no-migrations
+            [false, false],
+            [true, false],
+            [false, true],
+            [true, true],
```

Comment:
Tests should not be complex IMO, the expected return code should be here
```suggestion
            [false, false, 1],
            [true, false, 0],
            [false, true, 0],
            [true, true, 0],
```